### PR TITLE
[OV-DOC] Issues with pypi-openvino-dev.md

### PR DESCRIPTION
**Pull request by slyubimt**
### Details:
 - *Align virtual env instruction with openvino package install guide*
 - *Fix links to Kaldi and Caffe*
 - *Improve wording for tools descriptions* : See last comments: https://github.com/openvinotoolkit/openvino/pull/5756

### Tickets:
 - *57962*

### CODE REVIEW

File: docs/install_guides/pypi-openvino-dev.md [hunk 1]
```diff
@@ -54,17 +54,15 @@ To avoid dependency conflicts, use a virtual environment. Skip this
...
 
-On Linux and macOS:
-```sh
-# Depending on your OS, this step may require installing python3-venv
-python3 -m venv openvino_env
-```
-
-On Windows: 
 ```sh
+python -m pip install --user virtualenv
```
Comment (helena-intel) [inline docs/install_guides/pypi-openvino-dev.md:58]: Do you know if this is sufficient on a clean install of Ubuntu 18 and 20 (after installing pip, as in the note below), or do we have user acceptance testers that will test this on these systems?

File: docs/install_guides/pypi-openvino-dev.md [hunk 2]
```diff
@@ -87,11 +85,11 @@ python -m pip install --upgrade pip
 
 To install and configure the components of the development package for working with specific frameworks, use the `pip install openvino-dev[extras]` command, where `extras` is a list of extras from the table below: 
 
-| DL Framework                                                                     | Extra                          |
+| DL Framework                                                                     | Extra                           |
 | :------------------------------------------------------------------------------- | :-------------------------------|
 |   [Caffe*](https://caffe.berkeleyvision.org/)                                    | caffe                           |
-|   [Caffe2*](https://caffe2.ai/)                                                 | caffe2                          |
-|   [Kaldi*](https://kaldi-asr.org/)                                               | kaldi                           |
+|   [Caffe2*](https://github.com/pytorch/pytorch)                                  | caffe2                          |
```
Comment (helena-intel) [inline docs/install_guides/pypi-openvino-dev.md:91]: It is unexpected to see Caffe2 link to PyTorch.

Comment (slyubimt) [reply docs/install_guides/pypi-openvino-dev.md:90]: Caffe2 is part of PyTorch now. See deprecation notice here: https://caffe2.ai/

File: docs/install_guides/pypi-openvino-dev.md (hunk 1 above)
Comment (slyubimt) [reply docs/install_guides/pypi-openvino-dev.md:58]: Added instruction to install pip for Ubuntu

Comment (slyubimt) [reply docs/install_guides/pypi-openvino-dev.md:58]: Added instruction to install pip for Ubuntu.

**Comment by ncybulsk** [comment]
Validation passed

File: docs/install_guides/pypi-openvino-dev.md [hunk 3]
```diff
@@ -15,11 +15,11 @@ The **developer package** includes the following components installed by default
...
...el Downloader** is a tool for getting access to the collection of high-quality and extremely fast pre-trained deep learning [public](https://docs.openvinotoolkit.org/latest/omz_models_group_public.html) and [intel](https://docs.openvinotoolkit.org/latest/omz_models_group_intel.html)-trained models. These free pre-trained models can be used instead of training your own models, to speed up the development and production deployment process. The tool downloads model files from online sources and, if necessary, patches them to make them more usable with Model Optimizer. A number of additional tools are also provided to automate the process of working with downloaded models:<br> **Model Converter** is a tool for converting Open Model Zoo models that are stored in an original deep learning framework format into the Inference Engine Intermediate Representation (IR) using Model Optimizer. <br> **Model Quantizer** is a tool for automatic quantization of full-precision models in the IR format into low-precision versions using the Post-Training Optimization Tool. <br> **Model Information Dumper** is a helper utility for dumping information about the models to a stable, machine-readable format.
```
Comment (tsavina) [inline docs/install_guides/pypi-openvino-dev.md:22]:
```suggestion
| [Model Downloader and other Open Model Zoo tools](https://docs.openvinotoolkit.org/latest/omz_tools_downloader.html)| `omz_downloader` <br> `omz_converter` <br> `omz_quantizer` <br> `omz_info_dumper`| **Model Downloader** is a tool for getting access to the collection of high-quality and extremely fast pre-trained deep learning [public](https://docs.openvinotoolkit.org/latest/omz_models_group_public.html) and [Intel](https://docs.openvinotoolkit.org/latest/omz_models_group_intel.html)-trained models. These free pre-trained models can be used to speed up the development and production deployment process without training your own models. The tool downloads model files from online sources and, if necessary, patches them to make them more usable with Model Optimizer. A number of additional tools are also provided to automate the process of working with downloaded models:<br> **Model Converter** is a tool for converting Open Model Zoo models that are stored in an original deep learning framework format into the Inference Engine Intermediate Representation (IR) using Model Optimizer. <br> **Model Quantizer** is a tool for automatic quantization of full-precision models in the IR format into low-precision versions using the Post-Training Optimization Tool. <br> **Model Information Dumper** is a helper utility for dumping information about the models to a stable, machine-readable format.
```

**Review by andrew-zaytsev — APPROVED** [review]

**Pull request merged**